src/main/java/Duke.java

```diff
@@ -1,10 +1,153 @@
+
+    public static void bye() {
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static void checkCommand(String[] instructions) throws DukeException {
+        if (instructions.length == 0) {
+            throw new DukeException("Empty input.");
+        } else {
+            String command = instructions[0];
+            if (command.equals("bye") || command.equals("list")) {
+                if (instructions.length > 1) {
+                    throw new DukeException("There's too many input!");
+                }
+            } else if (command.equals("mark") || command.equals("unmark") || command.equals("delete")) {
+                if (instructions.length == 1) {
+                    throw new DukeException("Please specify the task number to " + command + ".");
+                }
+                int taskNumber = Integer.parseInt(instructions[1]);
+                if (taskNumber > taskList.size() || taskNumber <= 0) {
+                    throw new DukeException("Invalid task number! You have " + taskList.size() + " task(s).");
+                }
+            } else if (command.equals("todo")) {
+                if (instructions.length == 1) {
+                    throw new DukeException("Please specify the task you want to do.");
+                }
+            } else if (command.equals("deadline") || command.equals("event")) {
+                if (instructions.length == 1) {
+                    throw new DukeException("Please specify the description and time of the " + command + " you want to add.");
+                }
+                String[] taskAndTime = instructions[1].split("/");
+                if (taskAndTime.length < 2 || taskAndTime[0].equals("") || taskAndTime[1].length() < 3) {
+                    throw new DukeException("Invalid input. Please specify the description/time of the " + command + ".");
+                }
+            } else {
+                throw new DukeException("I don't understand your query.");
+            }
+        }
+    }
+
+    public static void chat() throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+
+        while (true) {
+            System.out.println(HORIZONTAL_LINE);
+            System.out.print("> ");
+            String userInput = br.readLine();
+            String[] instructions = userInput.split(" ", 2);
+            try {
+                checkCommand(instructions);
+                if (instructions[0].equals("bye")) {
+                    storage.writeToFile(taskList);
+                    break;
+                } else if (instructions[0].equals("list")) {
+                    list();
+                } else if (instructions[0].equals("mark") || instructions[0].equals("unmark") || instructions[0].equals("delete")) {
+                    modify(instructions);
+                } else {
+                    add(instructions);
+                }
```

Comment:
Following the Single Level of Abstraction Principle, since there is a `checkCommand` method, maybe it will be good to have a `processCommand` method as well?